save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/PermissionController.kt

```diff
@@ -116,7 +116,9 @@ class PermissionController(
             // fixme: could be `@PreAuthorize`, but organizationService cannot be found smh
             val userId = (authentication.details as AuthenticationDetails).id
             val hasOrganizationPermissions = organizationService.canChangeRoles(organizationName, userId)
-            val hasProjectPermissions = projectService.canChangeRoles(it, userId, setRoleRequest.userName, setRoleRequest.role)
+            val hasProjectPermissions =
+                    projectService.canChangeRoles(it, userId, setRoleRequest.userName, setRoleRequest.role)
+                        ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
```

Comment:
I think checking whether the user exists can be moved before the actual permissions check logic, e.g.:
```kotlin
...
.filter { project: Project ->
        // if project is hidden from the user, who attempts permission update,
        // then we should return 404
        projectPermissionEvaluator.hasPermission(authentication, project, Permission.READ)
    }
.zip(Mono.just( /* find user */ ))
.switchIfEmpty { /* if any of zipped Monos completes empty, the resulting Mono will copmlete empty */  }  
.filter { (project, user) ->
}
```
Then in ProjectService you should expect that `otherUser` exists.

Reply:
done